t/balancer.t

```diff
@@ -595,3 +595,169 @@ ok
 qr/\[error\] .*? log_by_lua.*? failed to call: API disabled in the current context/
 --- no_error_log
 [alert]
+
+
+
+=== TEST 14: set_timeout (connect_timeout, send_timeout, read_timeout)
+--- http_config
+    lua_package_path "$TEST_NGINX_CWD/lib/?.lua;;";
+
+    upstream backend {
+        server 0.0.0.1;
+        balancer_by_lua_block {
+            print("hello from balancer by lua!")
+            local b = require "ngx.balancer"
+            b.set_timeout(1000, 1000, 1000)
+            assert(b.set_current_peer("127.0.0.1", tonumber(ngx.var.server_port)))
+        }
+    }
+--- config
+    location = /t {
+        proxy_pass http://backend;
+    }
+    location = /back {
+        echo "hello proxy timeout!";
+    }
+--- request
+    GET /t
+--- response_body:
+hello proxy timeout!
+--- error_code: 200
+--- no_error_log
+[warn]
+
+
+
+=== TEST 15: set connect timeout zero got 504
+--- http_config
+    lua_package_path "$TEST_NGINX_CWD/lib/?.lua;;";
+
+    upstream backend {
+        server 0.0.0.1;
+        balancer_by_lua_block {
+            print("hello from balancer by lua!")
+            local b = require "ngx.balancer"
+            b.set_timeout(0, 1000, 1000)
```

Comment:
Hmm, zero timeout values are never useful (except for testing). We should test nonzero values like 1 (ms) instead.